tests/Doctrine/Tests/ORM/Proxy/ProxyFactoryTest.php

```diff
@@ -145,6 +146,38 @@ public function testFailedProxyLoadingDoesNotMarkTheProxyAsInitialized(): void
         self::assertFalse($proxy->__isInitialized());
     }
 
+    public function testExceptionOnProxyLoadingDoesNotMarkTheProxyAsInitialized(): void
+    {
+        $persister = $this
+            ->getMockBuilderWithOnlyMethods(BasicEntityPersister::class, ['load', 'getClassMetadata'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->uowMock->setEntityPersister(ECommerceFeature::class, $persister);
+
+        $proxy = $this->proxyFactory->getProxy(ECommerceFeature::class, ['id' => 42]);
+        assert($proxy instanceof Proxy);
+
+        $exception = $this
+            ->getMockBuilder(ConnectionException::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
Do we really need to mock an exception class?

Reply:
Well `ConnectionException` was the closest exception to the real-world scenario. And as it has a nasty constructor I have decided to mock it. It could be less complicated 👍 
And for the example: in my experience we have never encountered this with ORM, but with ODM and native `Client-Side Field Level Encryption` we had this issue. As our supervisor consumers catches all exceptions and keep working on messages. On `em->flush()` proxies that were not initialized has been flushed to db removing all fields that had no default values.
After I made PR to ODM lib decided to check if the same is possible with ORM also. And theoretically it is possible.